src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6543,6 +6554,33 @@ bool CracRestoreParameters::read_from(int fd) {
   ::_restore_start_time = hdr->_restore_time;
   ::_restore_start_counter = hdr->_restore_counter;
 
+  for (int i = 0; i < hdr->_nflags; i++) {
+    FormatBuffer<80> err_msg("%s", "");
+    JVMFlag::Error result;
+    const char *name = cursor;
+    if (*cursor == '+' || *cursor == '-') {
+      name = cursor + 1;
+      result = WriteableFlags::set_flag(name, *cursor == '+' ? "true" : "false",
+        JVMFlagOrigin::CRAC_RESTORE, err_msg);
+      cursor += strlen(cursor) + 1;
+    } else {
+      char* eq = strchrnul(cursor, '=');
+      if (*eq == '\0') {
+        result = JVMFlag::Error::MISSING_VALUE;
+        cursor = eq + 1;
+      } else {
+        *eq = '\0';
+        char* value = eq + 1;
+        result = WriteableFlags::set_flag(cursor, value, JVMFlagOrigin::CRAC_RESTORE, err_msg);
+        cursor = value + strlen(value) + 1;
+      }
+    }
+    if (result != JVMFlag::Error::SUCCESS) {
+      warning("VM Option '%s' cannot be changed, ignoring: %s",
```

Comment:
A significant set of options cannot be set on restore at the moment, so it will be even better to highlight they don't have effect and produce an error. It may be useful to revert back to warning (with e.g. an option), but by default it should be disabled (leading to the error)

Reply:
The place can be `guarantee(result == JVMFlag::Error::SUCCESS, "...")` since the possibility was checked earlier.